index.d.ts

```diff
@@ -0,0 +1,2148 @@
+ * Once created a Shipment object is used to retrieve shipping Rates and purchase a label.
+ * 
+ * A Shipment created with a valid to_address, from_address, and parcel will automatically populate its rates attribute.
+ * 
+ * @see https://www.easypost.com/docs/api/node#shipment-object
+ */
+export declare interface IShipment extends IObjectWithId<"Shipment">, IDatedObject {
+  /**
+   * An optional field that may be used in place of id in other API endpoints
+   */
+  reference?: string;
+
+  /**
+   * The destination address
+   */
+  to_address: IAddress;
+
+  /**
+   * The origin address
+   */
+  from_address: IAddress;
+
+  /**
+   * The shipper's address, defaults to from_address
+   */
+  return_address: IAddress;
+
+  /**
+   * The buyer's address, defaults to to_address
+   */
+  buyer_address: IAddress;
+
+  /**
+   * The dimensions and weight of the package
+   */
+  parcel: IParcel;
+
+  /**
+   * Information for the processing of customs
+   */
+  customs_info: ICustomsInfo;
+
+  /**
+   * Document created to manifest and scan multiple shipments
+   */
+  scan_form: IScanForm;
+
+  /**
+   * All associated Form objects
+   */
+  forms: IForm[];
+
+  /**
+   * The associated Insurance object
+   */
+  insurance: IInsurance;
+
+  /**
+   * All associated Rate objects
+   */
+  rates: IRate[];
+
+  /**
+   * The specific rate purchased for the shipment, or null if unpurchased or purchased through another mechanism
+   */
+  selected_rate: IRate;
+
+  /**
+   * The associated PostageLabel object
+   */
+  postage_label: IPostageLabel;
+
+  /**
+   * Any carrier errors encountered during rating, discussed in more depth below
+   */
+  messages: IMessage[];
+
+  /**
+   * All of the options passed to the shipment, discussed in more depth below
+   */
+  options: IOptions;
+
+  /**
+   * Set true to create as a return, discussed in more depth below
+   */
+  is_return: boolean;
+
+  /**
+   * If purchased, the tracking code will appear here as well as within the Tracker object
+   */
+  tracking_code: string;
+
+  /**
+   * The USPS zone of the shipment, if purchased with USPS
+   */
+  usps_zone: string;
+
+  /**
+   * The current tracking status of the shipment
+   */
+  status: string;
+
+  /**
+   * The associated Tracker object
+   */
+  tracker: ITracker;
+
+  /**
+   * The associated Fee objects charged to the billing user account
+   */
+  fees: IFee[];
+
+  /**
+   * The current status of the shipment refund process. Possible values are "submitted", "refunded", "rejected".
+   */
+  refund_status: "submitted" | "refunded" | "rejected";
+
+  /**
+   * The ID of the batch that contains this shipment, if any
+   */
+  batch_id: string;
+
+  /**
+   * The current state of the associated BatchShipment
+   */
+  batch_status: TBatchStatus; // TODO state or status? (Description says state but property says status)
```

Comment:
state or status? [Description](https://www.easypost.com/docs/api/node#shipment-object) says state but property says status

Reply:
On a Batch object, there is both `state` and `status`: https://www.easypost.com/docs/api/node#batch-object

Reply:
Sorry if this seems nitpicky :sweat_smile:

Just some minor confusion on my part while reading the docs. No code changes need to be made here. I think this could be resolved with a small edit to the docs.

___

The [Shipment Object](https://www.easypost.com/docs/api/node#shipment-object) has a property `batch_status` but the description for that property (on the [Shipment Object](https://www.easypost.com/docs/api/node#shipment-object) ) says:

> The current `state` of the associated [BatchShipment](https://www.easypost.com/docs/api/node#batch-shipment-object)

I think this should actually say the following:

> The current `status` of the associated [BatchShipment](https://www.easypost.com/docs/api/node#batch-shipment-object)

as the [BatchShipment Object](https://www.easypost.com/docs/api/node#batch-shipment-object) doesn't have a `state` - just a "status" (`batch_status`)

![image](https://user-images.githubusercontent.com/2395597/157882300-7281ba6c-4f5e-43e3-97df-32bb9ed0e95c.png)

![image](https://user-images.githubusercontent.com/2395597/157886687-193958cf-386e-4787-acfa-f08e2c02e833.png)

Reply:
Fair enough, I can tweak this one too.